src/sharding.js

```diff
@@ -128,52 +127,45 @@ class ShardingDatastore {
   }
 
   query (q /* : Query<Buffer> */) /* : QueryResult<Buffer> */ {
-    const tq/* : Query<Buffer> */ = {
-      keysOnly: q.keysOnly
+    let tq/* : Query<Buffer> */ = {
+      keysOnly: q.keysOnly,
+      offset: q.offset,
+      limit: q.limit,
+      filters: [
+        (e, cb) => cb(null, e.key.toString() !== shardKey.toString()),
+        (e, cb) => cb(null, e.key.toString() !== shardReadmeKey.toString())
+      ]
     }
 
     if (q.prefix != null) {
-      // TODO: transform
-      tq.prefix = q.prefix
+      tq.filters.push((e, cb) => {
+        cb(null, this._invertKey(e.key).toString().startsWith(q.prefix))
+      })
     }
 
     if (q.filters != null) {
-      tq.filters = q.filters.map((f) => (e, cb) => {
+      let filters = q.filters.map((f) => (e, cb) => {
```

Comment:
/let/const/

Reply:
got me, consistency is the hobglobin of little minds!

I'll make both changes